createtemplates.sh

```diff
@@ -0,0 +1,327 @@
+    #"terraform/aws/101-caas-ecs"
+    #"terraform/aws/101-genai-ecs"
+    #"terraform/aws/101-genai-eks"
+    #"terraform/aws/101-kubeflow-eks"
+    #"terraform/aws/101-naas-eks"
+    #"terraform/aws/101-vpc-ec2"
+    #"terraform/aws/101-vpc-ecs"
+    #"terraform/aws/101-waas-eks"
+    #"terraform/azure/101-vnet-instance"
+    #"terraform/gcp/101-caas-gke"
+    #"terraform/gcp/101-vpc-instance"
+    #"terraform/mks/101-caas-pnap"
+    #"terraform/oci/101-caas-oke"
+    #"terraform/vcluster/101-caas"
+    #"terraform/vmware/101-caas-vmware"
+    #"terraform/waas/101-waas"
+)
+
+# POST request for HUB apis
+function make_get_request_new {
+    local endpoint="$1"
+    curl -s -X GET -H "Accept */*" -H "X-API-KEY: ${API_KEY}" "${BASE_URL}/${endpoint}"
+}
+
+# POST request for old apis
+function make_get_request_old {
+    local endpoint="$1"
+    curl -s --location "${BASE_URL}/${endpoint}" -H "X-RAFAY-API-KEYID: ${API_KEY}"
+}
+
+# POST request for HUB apis
+function make_post_request_new {
+    local endpoint="$1"
+    local data_file="$2"
+    echo "${BASE_URL}/${endpoint}"
+    curl -s -X POST -H "Content-Type: application/json" -H "X-API-KEY: ${API_KEY}" -d "@${data_file}" "${BASE_URL}/${endpoint}" 
+}
+
+# POST request for old apis
+function make_post_request_old {
+    local endpoint="$1"
+    local data_file="$2"
+    echo "${BASE_URL}/${endpoint}"
+    curl -s -X POST -H "Content-Type: application/json" -H "X-RAFAY-API-KEYID: ${API_KEY}" -d "@${data_file}" "${BASE_URL}/${endpoint}" 
+}
+
+# creates configcontext templates
+function create_project {
+    curl -s -X POST -H "Content-Type: application/json" -H "X-RAFAY-API-KEYID: ${API_KEY}" -d "{\"name\":\"${PROJECT_NAME}\",\"description\":\"Project for templates\"}" "${BASE_URL}/$CREATE_PROJECT_URL/" | jq > project_response.json
+    PROJECT_HASH=$(cat project_response.json | jq -r '.id')
+    echo "Successfully created project ${PROJECT_NAME} id ${PROJECT_HASH}"
+}
+
+# creates configcontext templates
+function create_configcontext_templates {
+    local templatefile="$1"
+    local folder="$2"
+    echo "Creating you configcontext for ${templatefile}"
+
+    sed "s/^$PROJECT_NAME_FIELD: .*$/$PROJECT_NAME_FIELD: $PROJECT_NAME/" $PWD/$folder/setup/values.yaml > $PWD/$folder/setup/tmp.yaml
+    
+    fName="$(rctl apply -t $templatefile --values $PWD/$folder/setup/tmp.yaml \
+           --test-template | grep -A1 metadata | awk  '/name/ {print $2}')" 
+
+    ##Generate spec using rctl
+    if ! rctl apply -t $templatefile --values $PWD/$folder/setup/tmp.yaml \
+        --test-template --write $PWD/$folder/setup/templates/$fName.yaml.spec ; then
+        printf -- "\033[31m ERROR: Rafay spec file for %s failed - FAILED \033[0m\n" "$i";
+        exit
+    else
+        printf -- "\033[32m Info: Rafay spec file for %s - SUCCESS \033[0m\n" "$i";
+    fi
+    
+    yq -o=json $PWD/$folder/setup/templates/$fName.yaml.spec > $PWD/$folder/setup/templates/$fName.json 
```

Comment:
This script requires few external tools, Can we add something like this here as well?
 https://github.com/hardik-rafay/envmgr/blob/main/terraform/aws/101-caas-ecs/setup/setup.sh#L9

Reply:
done